contracts/FungibleTokenSwitchboard.cdc

```diff
@@ -0,0 +1,151 @@
+import FungibleToken from "./FungibleToken.cdc"
+
+/// FungibleTokenSwitchboard
+///
+/// The contract that allows an account to receive payments in multiple fungible
+/// tokens using a single {FungibleToken.Receiver}
+///
+pub contract FungibleTokenSwitchboard {
+    
+    // Storage and Public Paths
+    pub let SwitchboardStoragePath: StoragePath
+    pub let SwitchboardPublicPath: PublicPath
+    pub let SwitchboardReceiverPublicPath: PublicPath
+    
+    /// FungibleTokenSwitchboardInitialized
+    ///
+    /// The event emitted when the contract is created
+    ///
+    pub event FungibleTokenSwitchboardInitialized()
+    
+    /// SwitchboardInitialized
+    ///
+    /// The event that is emited when a new Switchboard resource is created
+    ///
+    pub event SwitchboardInitialized(switchboardResourceID: UInt64)
+
+    /// VaultCapacityAdded
+    ///
+    /// The event that is emitted when a new vault capacity is added to a
+    /// switchboard resource
+    ///
+    pub event VaultCapabilityAdded(type: Type)
+
+    /// VaultCapacityRemoved
+    ///
+    /// The event that is emitted vault capacity is added removed from a 
+    /// switchboard resource
+    ///
+    pub event VaultCapabilityRemoved(type: Type)
+    
+    /// SwitchboardPublic
+    ///
+    /// The interface that enforces the method to allow anyone to check on the
+    /// available capabilities of a switchboard resource and also exposes the 
+    /// deposit method to deposit funds on it
+    ///
+    pub resource interface SwitchboardPublic {
+        pub fun getVaultCapabilities(): {Type: Capability<&{FungibleToken.Receiver}>}
+        pub fun deposit(from: @FungibleToken.Vault)
+    }
+    
+    /// Switchboard
+    /// The resource that stores the multiple fungible token receiver capabilities,
+    /// allowing the owner to add and remove then and anyone to deposit any
+    /// fungible token among the available capabilities
+    ///
+    pub resource Switchboard: FungibleToken.Receiver, SwitchboardPublic {
+        
+        /// fungibleTokenReceiverCapabilities
+        /// Dictoionary holding the fungible token receiver capabilities, 
+        /// indexed by the fungible token vault type
+        ///
+        pub var fungibleTokenReceiverCapabilities: {Type: Capability<&{FungibleToken.Receiver}>}
```

Comment:
Yeah I did that on purpose I hated to see every variable pub and some access(contract) in dictionarys / arrays, but I agree, since we have a getter there's no real need of having it public